tangelo/algorithms/classical/fci_solver.py

```diff
@@ -131,3 +144,170 @@ def get_rdm(self):
+        return energy
+
+    def get_rdm(self):
+        """Compute the Full CI 1- and 2-particle reduced density matrices.
+
+        Returns:
+            numpy.array: One-particle RDM.
+            numpy.array: Two-particle RDM.
+
+        Raises:
+            RuntimeError: If method "simulate" hasn't been run.
+        """
+
+        # Check if Full CI is performed
+        if self.ciwfn is None:
+            raise RuntimeError("FCISolver: Cannot retrieve RDM. Please run the 'simulate' method first")
+        if self.any_close:
+            warnings.warn("Degenerate orbitals are present in the molecule. The fci calculation is performed "
+                          "without symmetry so the rdms may not correspond to the integrals in molecule. A c1 "
+                          "version of the molecule with the correct integrals is present as FCISolverPsi4.molecule.")
+        one_rdm = np.asarray(self.ciwfn.get_opdm(0, 0, "SUM", False))
+        two_rdm = np.asarray(self.ciwfn.get_tpdm("SUM", True))
+
+        return one_rdm, two_rdm
+
+
+fci_solver_dict = {"pyscf": FCISolverPySCF, 'psi4': FCISolverPsi4}
+
+
+def get_fci_solver(molecule: SecondQuantizedMolecule, solver: Union[None, str, Type[ElectronicStructureSolver]] = default_fci_solver, **kwargs):
+    """Return requested target backend object.
+
+    Args:
+        solver (string or Type[ElectronicStructureSolver] or None): Supported string identifiers can be found in
+            fci_solver_dict (from tangelo.algorithms.classical.fci_solver). Can also provide a user-defined backend (child to ElectronicStructureSolver class)
+        kwargs: Other arguments that could be passed to a target. Examples are qubits_to_use for a QPU, transpiler
+            optimization level, error mitigation flags etc.
+     """
+
+    if solver is None:
+        if isinstance(molecule.solver, IntegralSolverPySCF):
+            solver = FCISolverPySCF
+        elif isinstance(molecule.solver, IntegralSolverPsi4):
+            solver = FCISolverPsi4
+        elif default_fci_solver is not None:
+            solver = default_fci_solver
+        else:
+            raise ModuleNotFoundError(f"One of the backends for {list(fci_solver_dict.keys())} needs to be installed to use FCISolver in Tangelo"
+                                      "without providing own implementation.")
+    # If target is a string use target_dict to return built-in backend
+    elif isinstance(solver, str):
+        try:
+            solver = fci_solver_dict[solver.lower()]
+        except KeyError:
+            raise ValueError(f"Error: backend {solver} not supported. Available built-in options: {list(fci_solver_dict.keys())}")
+    elif not issubclass(solver, ElectronicStructureSolver):
+        raise TypeError(f"Target must be a str or a subclass of Backend but received class {type(solver).__name__}")
+
+    return solver(molecule, **kwargs)
+
+
+# @deprecated("Please use get_fci_solver.")
```

Comment:
Any reason we should use `get_fci_solver` over `FCISolver` ? It seems to be just a renaming to adhere to PEP8.

We did that for the quantum simulation backends because so many choices (and therefore Backend class names) and it makes sense for developers and users to loop over them or easily switch. I do not remember if there was another reason.

But in this context, what do you dislike about people writing `fci_solver = FCISolver(mol, "pyscf")` ?

Reply:
We went back and forth many times for Simulator. My understanding was that we changed to `get_backend` mostly for PEP8 reasons. I would prefer to keep both but not use deprecate.